aten/src/ATen/native/quantized/cpu/BinaryOps.cpp

```diff
@@ -393,25 +393,19 @@ Tensor xnnp_add(Tensor qa, Tensor qb, double scale, int64_t zero_point) {
 template <bool ReLUFused = false>
 Tensor qadd(Tensor qa, Tensor qb, double scale, int64_t zero_point) {
   check_inputs(qa, qb);
-
-  if (at::globalContext().qEngine() == at::QEngine::QNNPACK) {
-    TORCH_CHECK(
-        qa.scalar_type() == qb.scalar_type(),
-        "Both inputs to qadd must have same type");
-
 #ifdef USE_XNNPACK
-    if (qa.scalar_type() == kQInt8) {
-          return xnnp_add<c10::qint8, ReLUFused>(qa, qb, scale, zero_point);
-    }
+  if (qa.scalar_type() == kQInt8) {
```

Comment:
checking this might be sufficient given other qengines fbgemm, x86 etc. should have uint8 input but curious why move it out of the check, how are you quantizing this op?

Reply:
This is perhaps no longer relevant (see above answer).